python/tskit/trees.py

```diff
@@ -2338,6 +2340,113 @@ def kc_distance(self, other, lambda_=0.0):
+        :param float span: The span of the tree, and therefore the
+            :attr:`~TreeSequence.sequence_length` of the :attr:`~Tree.tree_sequence`
+            property of the returned :class:<Tree>
+        :param float branch_length: The length of every branch in the tree (equivalent
+            to the time of the root node).
+
+        .. note::
+            This is similar to ``tskit.Tree.unrank((0,0), n, span=span)`` but is more
+            efficient for large n. However, the ``unrank`` method provides
+            a concise way of generating alternative (non-star) topologies.
+        """
+        tc = tables.TableCollection(sequence_length=span)
+        tc.nodes.set_columns(
+            flags=np.full(n, NODE_IS_SAMPLE, dtype=np.uint32),
+            time=np.zeros(n),
+        )
+        if n > 1:
+            root = tc.nodes.add_row(time=branch_length)
+            tc.edges.set_columns(
+                left=np.full(n, 0),
+                right=np.full(n, span),
+                parent=np.full(n, root, dtype=np.int32),
+                child=np.arange(n, dtype=np.int32),
+            )
+        if record_provenance:
+            # TODO replace with a version of https://github.com/tskit-dev/tskit/pull/243
+            # TODO also make sure we convert all the arguments so that they are
+            # definitely JSON encodable.
+            parameters = {"command": "create_star", "TODO": "add parameters"}
+            tc.provenances.add_row(
+                record=json.dumps(provenance.get_provenance_dict(parameters))
+            )
+        return tc.tree_sequence().first()
+
+    @staticmethod
+    def create_balanced(n, *, span=1, branch_length=1, record_provenance=True):
+        """
+        Create a single bifurcating :class:<Tree> and associated tree sequence where
+        the tree is a completely balanced (i.e. all parents have exactly 2 children,
+        and all leaves go through the the same number of nodes to get to the root).
+
+        :param int n: The number of leaf nodes in the tree (must be an integer
+            power of 2).
+        :param float span: The span of the tree, and therefore the
+            :attr:`~TreeSequence.sequence_length` of the :attr:`~Tree.tree_sequence`
+            property of the returned :class:<Tree>
+        :param float branch_length: The length of every branch in the tree.
+
+        .. note::
+            This creates a tree topology identical to that generated by
+            ``tskit.Tree.unrank((tskit.combinatorics.num_shapes(n)-1, 0), n, span=span)``
+            but is more efficient for large n. However, the ``unrank`` method provides
+            a concise way of generating alternative (non-balanced) topologies.
+        """
+        tc = tables.TableCollection(sequence_length=span)
+        tc.nodes.set_columns(
+            flags=np.full(n, NODE_IS_SAMPLE, dtype=np.uint32),
+            time=np.zeros(n),
+        )
+        if n & (n - 1) != 0:
```

Comment:
Seems a bit lame to insist ``num_leaves`` is a power of 2. Why not make the tree as balanced as possible? It'll be the same algorithm I'd imagine. It would be nice if this was also identical to ``tskit.Tree.unrank(N - 1, 0)``

Reply:
Well, then the tree isn't actually balanced. So if anyone was to run it to test e.g. measures of tree balance, then they would fail. Given the name of the function, I think we should insist on creating a fully balanced tree, unless e.g. there is a flag "allow_partially_unbalanced" (default False) so that users have to say that they are completely sure that's what they want.

Reply:
It's as balanced as possible for the given number of leaves though. This will basically be useless if we only allow powers of 2. What do we get for ``unrank(N - 1)`` for num_leaves = 5, say?

Reply:
> > I agree it's nice to have all these have a common prefix. I'm not so sure about "create_" though; perhaps "make_"?
> 
> Or `generate`? Happy with any of those.

Just realised that "make_star" might imply that we are forcing an existing tree into a star shape. So if you don't like "create", I would prefer "generate". Perhaps @benjeffery might care to adjudicate?

Reply:
> Perhaps @benjeffery might care to adjudicate?

Happy with `make` or `generate`, with a very slight preference for `generate`.